sdk/data/azcosmos/cosmos_global_endpoint_manager.go

```diff
@@ -0,0 +1,170 @@
+package azcosmos
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sync"
+	"time"
+
+	azlog "github.com/Azure/azure-sdk-for-go/sdk/azcore/log"
+	azruntime "github.com/Azure/azure-sdk-for-go/sdk/azcore/runtime"
+	"github.com/Azure/azure-sdk-for-go/sdk/internal/log"
+)
+
+const DefaultBackgroundRefreshInterval = 5 * time.Minute
+
+type globalEndpointManager struct {
+	client           *Client
+	preferredRegions []string
+	locationCache    *locationCache
+	refreshTicker    *time.Ticker
+	refreshDone      chan struct{}
+	sync.Mutex
+}
+
+// NewGlobalEndpointManager creates a new global endpoint manager.
+func NewGlobalEndpointManager(client *Client, preferredRegions []string) (*globalEndpointManager, error) {
+	endpoint, err := url.Parse(client.endpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return &globalEndpointManager{
+		client:           client,
+		preferredRegions: preferredRegions,
+		locationCache:    newLocationCache(preferredRegions, *endpoint),
+	}, nil
+}
+
+// GetWriteEndpoints returns the write endpoints from the location cache.
+func (gem *globalEndpointManager) GetWriteEndpoints() ([]url.URL, error) {
+	return gem.locationCache.writeEndpoints()
+}
+
+// GetReadEndpoints returns the read endpoints from the location cache.
+func (gem *globalEndpointManager) GetReadEndpoints() ([]url.URL, error) {
+	return gem.locationCache.readEndpoints()
+}
+
+// GetAccountProperties retrieves account properties from the Cosmos DB instance.
+func (gem *globalEndpointManager) GetAccountProperties() (accountProperties, error) {
+	operationContext := pipelineRequestOptions{
+		resourceType:    resourceTypeDatabaseAccount,
+		resourceAddress: "",
+	}
+
+	path, err := generatePathForNameBased(resourceTypeDatabaseAccount, "", false)
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to generate path for name-based request: %w", err)
+	}
+
+	azResponse, err := gem.client.sendGetRequest(path, context.Background(), operationContext, nil, nil)
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to send GET request: %w", err)
+	}
+
+	properties, err := parseAccountProperties(azResponse)
+	if err != nil {
+		return accountProperties{}, fmt.Errorf("failed to parse account properties: %w", err)
+	}
+
+	return properties, nil
+}
+
+// parseAccountProperties parses the account properties from the HTTP response.
+func parseAccountProperties(azResponse *http.Response) (accountProperties, error) {
```

Comment:
Because this method creates an accountProperties object, the ideal naming convention would be:

```suggestion
func newAccountProperties(azResponse *http.Response) (accountProperties, error) {
```